### PR TITLE
[NativeAOT] Miscellaneous cleanup

### PR DESCRIPTION
**Pull request by jkotas**
- Delete unnecessary abstractions
- Simplify ifdefs
- Use more derived types

### CODE REVIEW

**Comment by jkotas** [comment]
Peeled off from #93440.

**Comment by jkotas** [comment]
@MichalStrehovsky PTLA

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/RuntimeTypeUnifier.cs [hunk 1]
```diff
@@ -42,12 +43,12 @@ internal static partial class RuntimeTypeUnifier
         //
         // Retrieves the unified Type object for given RuntimeTypeHandle (this is basically the Type.GetTypeFromHandle() api without the input validation.)
         //
-        internal static Type GetRuntimeTypeForEEType(EETypePtr eeType)
+        internal static unsafe RuntimeType GetRuntimeTypeForMethodTable(MethodTable* eeType)
         {
             // If writable data is supported, we shouldn't be using the hashtable - the runtime type
             // is accessible through a couple indirections from the EETypePtr which is much faster.
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/RuntimeTypeUnifier.cs:49]:
```suggestion
            // is accessible through a couple indirections from the MethodTable which is much faster.
```

File: src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs [hunk 2]
```diff
@@ -259,12 +259,7 @@ public override bool IsEnumDefined(object value)
         public override bool IsInstanceOfType([NotNullWhen(true)] object? o) => RuntimeTypeHandle.IsInstanceOfType(this, o);
 
         public override bool IsAssignableFrom([NotNullWhen(true)] TypeInfo? typeInfo)
-        {
-            if (typeInfo == null)
-                return false;
-
-            return IsAssignableFrom(typeInfo.AsType());
-        }
+            => (typeInfo == null) ? false : IsAssignableFrom(typeInfo.AsType());
```
Comment (MichalStrehovsky) [inline src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs:262]:
```suggestion
            => typeInfo != null && IsAssignableFrom(typeInfo.AsType());
```

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**